prow/pod-utils/gcs/upload_test.go

```diff
@@ -58,6 +57,8 @@ func TestUploadToGcs(t *testing.T) {
 		},
 	}
 
+	retriesWhenFailing := 3
+
 	for _, testCase := range testCases {
```

Comment:
Please add a case that validates we will retry successfully.

AKA add something like a `flaky := func(o *storage.OH) error { *count++; if count != 3 { return err }}`